Assets/Scripts/PathMovement.cs

```diff
@@ -76,28 +80,39 @@ public class PathMovement : MonoBehaviour {
     // 1. there is a cap to path length
     // 2. the movement is uniform speed no matter the path length
     public void moveToNextPointOnPath() {
-        if(nextPointIndex < 0) {
+        // check for point to move to;
+        if(nextPointIndex < 0)
             return;
-        }
+
+        // get next point
         Vector3 nextPoint;
-        try {
+        try 
+        {
             nextPoint = path[nextPointIndex];
-
-        } catch(System.ArgumentOutOfRangeException e){
+        }
+        catch(System.ArgumentOutOfRangeException e)
+        {
             return;
         }
-        
-        var maxDistanceDelta = Time.deltaTime*speed;
-       // thingToMove.transform.LookAt(nextPoint);
-        thingToMove.transform.position = Vector3.MoveTowards(thingToMove.transform.position, nextPoint, maxDistanceDelta);
-       // Debug.Log(Vector3.Distance(thingToMove.transform.position, nextPoint) + " " + maxDistanceDelta);
-        if(Vector3.Distance(thingToMove.transform.position, nextPoint) < maxDistanceDelta) {
-            if (nextPointIndex > 0) {
-                path.RemoveAt(nextPointIndex - 1);
-                nextPointIndex--;
+
+        // move
+        if(staminaBar.move())
+        {
```

Comment:
Is this intended behavior? The player will abruptly stops when it runs out of stamina, even though the path seems to indicate that it can move further. You might need to limit the path length to reflect how much stamina left